src/lib/mixer/MixerGroup.hpp

```diff
@@ -0,0 +1,170 @@
+ *
+ ****************************************************************************/
+
+#pragma once
+
+#include "Mixer.hpp"
+
+/**
+ * Group of mixers, built up from single mixers and processed
+ * in order when mixing.
+ */
+class MixerGroup
+{
+public:
+	MixerGroup() = default;
+
+	~MixerGroup()
+	{
+		reset();
+	}
+
+	// no copy, assignment, move, move assignment
+	MixerGroup(const MixerGroup &) = delete;
+	MixerGroup &operator=(const MixerGroup &) = delete;
+	MixerGroup(MixerGroup &&) = delete;
+	MixerGroup &operator=(MixerGroup &&) = delete;
+
+	unsigned			mix(float *outputs, unsigned space);
+
+	uint16_t			get_saturation_status();
+
+	void				groups_required(uint32_t &groups);
+
+	/**
+	 * Add a mixer to the group.
+	 *
+	 * @param mixer			The mixer to be added.
+	 */
+	void				add_mixer(Mixer *mixer) { _mixers.add(mixer); }
+
+	/**
+	 * Remove all the mixers from the group.
+	 */
+	void				reset() { _mixers.clear(); }
+
+	/**
+	 * Count the mixers in the group.
+	 */
+	unsigned			count() const { return _mixers.size(); }
+
+	/**
+	 * Adds mixers to the group based on a text description in a buffer.
+	 *
+	 * Mixer definitions begin with a single capital letter and a colon.
+	 * The actual format of the mixer definition varies with the individual
+	 * mixers; they are summarised here, but see ROMFS/mixers/README for
+	 * more details.
+	 *
+	 * Null Mixer
+	 * ..........
+	 *
+	 * The null mixer definition has the form:
+	 *
+	 *   Z:
+	 *
+	 * Simple Mixer
+	 * ............
+	 *
+	 * A simple mixer definition begins with:
+	 *
+	 *   M: <control count>
+	 *   O: <-ve scale> <+ve scale> <offset> <lower limit> <upper limit>
+	 *
+	 * The second line O: can be omitted. In that case 'O: 10000 10000 0 -10000 10000' is used.
+	 * The definition continues with <control count> entries describing the control
+	 * inputs and their scaling, in the form:
+	 *
+	 *   S: <group> <index> <-ve scale> <+ve scale> <offset> <lower limit> <upper limit>
+	 *
+	 * Multirotor Mixer
+	 * ................
+	 *
+	 * The multirotor mixer definition is a single line of the form:
+	 *
+	 * R: <geometry> <roll scale> <pitch scale> <yaw scale> <deadband>
+	 *
+	 * Helicopter Mixer
+	 * ................
+	 *
+	 * The helicopter mixer includes throttle and pitch curves
+	 *
+	 * H: <swash plate servo count>
+	 * T: <0> <2500> <5000> <7500> <10000>
+	 * P: <-10000> <-5000> <0> <5000> <10000>
+	 *
+	 * The definition continues with <swash plate servo count> entries describing
+	 * the position of the servo, in the following form:
+	 *
+	 *   S: <angle (deg)> <normalized arm length> <scale> <offset> <lower limit> <upper limit>
+	 *
+	 * @param buf			The mixer configuration buffer.
+	 * @param buflen		The length of the buffer, updated to reflect
+	 *				bytes as they are consumed.
+	 * @return			Zero on successful load, nonzero otherwise.
+	 */
+	int				load_from_buf(Mixer::ControlCallback control_cb, uintptr_t cb_handle, const char *buf, unsigned &buflen);
```

Comment:
You may implement instead
```
static MixerGroup *from_text(Mixer::ControlCallback control_cb, uintptr_t cb_handle, const char *buf, unsigned &buflen);
```
to be fully consistent with the other mixers?

Reply:
Let's think about that for a follow up. MixerGroup is now just the thing that you go through to interact with the full set of loaded mixers. It's no longer a Mixer.